src/structures/GuildChannel.js

```diff
@@ -68,13 +71,8 @@ class GuildChannel extends Channel {
     }
 
     if ('permission_overwrites' in data) {
-      /**
-       * A map of permission overwrites in this channel for roles and users
-       * @type {Collection<Snowflake, PermissionOverwrites>}
-       */
-      this.permissionOverwrites = new Collection();
```

Comment:
I can't exactly test this thesis now due the issue pointed out by my above comment, but the manager's cache should be `clear`-ed here to actually remove removed overwrites.